libs/mobilePayments.js

```diff
@@ -0,0 +1,52 @@
+const nconf = require('nconf');
+const request = require('request');
+const moment = require('moment');
+
+const BASE_URL = nconf.get('BASE_URL');
+
+api.cancelSubscriptionForUser = function cancelSubscriptionForUser (habitrpgUsers, user, platform) {
+    return new Promise((resolve, reject) => {
+      request.get(`${BASE_URL}/iap/${patform}/subscribe/cancel`, {
+      qs: {
+        noRedirect: 'true',
+      },
+      headers: {
+        'x-api-user': user._id,
+        'x-api-key': user.apiToken,
+      },
+      }, (habitError, habitResponse, body) => {
+          if (habitResponse.statusCode == 401) {
```

Comment:
ugh, it suuucks that the API returns 401 here. It's not the worker server's fault that we make such inappropriate use of our HTTP status codes, but doesn't this mean we'll end up forcing subscriptions closed if we hit a 401 for some other reason? I guess we wouldn't e.g. run this against a server with basic auth, but... is there anything else in the response we can look at that'd be better targeted?

Also, should be `===` not `==`

Reply:
I'd say the 401 is appropriate in the case of a blocked user.
We have two options for resolving this:

1. Allowing the API call for cancelling a sub to work even if the user is blocked (not sure how much work this would be since I'm not sure if there is precedence that an API call is excluded from this
2. add a computer readable error code/message that we can check